pkg/kudoctl/kudoinit/setup/wait.go

```diff
@@ -24,30 +24,42 @@ import (
 // Returns true if it exists. If the timeout was reached and it could not find the pod, it returns false.
 func WatchKUDOUntilReady(client kubernetes.Interface, opts kudoinit.Options, timeout int64) error {
 	return wait.PollImmediate(500*time.Millisecond, time.Duration(timeout)*time.Second,
-		func() (bool, error) { return verifyKudoDeployment(client.CoreV1(), opts.Namespace, opts.Image) })
+		func() (bool, error) { return verifyKudoDeployment(client.CoreV1(), opts.Namespace) })
 }
 
-func verifyKudoDeployment(client corev1.PodsGetter, namespace, expectedImage string) (bool, error) {
-	image, err := getKUDOPodImage(client, namespace)
-	if err == nil && image == expectedImage {
+func verifyKudoDeployment(client corev1.PodsGetter, namespace string) (bool, error) {
+	ready, err := isKUDOPodReady(client, namespace)
+	if err == nil && ready {
```

Comment:
Not that I object but why was the image comparison removed?

Reply:
good question... I didn't see any value in it

Reply:
I think the value was being able to establish whether it's the version of pod we expect. Not useful when the pod is being created initially, but when we are rolling out an update to the manager deployment, we need to guard against declaring victory after just checking that the pre-existing pod is healthy, i.e. the following race scenario:
- apply statefulset change
- statefulset controller is lazy, doing nothing yet
- we check the (old!) pod health, hey! it's healthy, consider the rollout done
- only now does the statefulset controller replace the pod with the new one...

Reply:
in agreement with the need for awareness of this scenario... 
1. currently we do NOT apply a change to statefulset... we delete, then apply
2. I'm not confident that image version change is sufficient... for instance, is it possible that a user may want to "upgrade" to the latest service-accounts, crds, etc, but use a previous version of kudo manager?   The version is a flag controllable thing... Assuming it is the same version of the cli is likely the 80% use case... perhaps more.. but doesn't capture all possibilities.
3. it would be better IMO to either wait for deletes to complete... or capture the current statefulset prior to upgrade and check revision.  I'll create an Issue to capture this.

Reply:
:100: